src/troute-network/troute/nhd_io.py

```diff
@@ -1948,14 +1948,104 @@ def write_waterbody_netcdf(
+                
+                # =========== q,v,d,ndg VARIABLES ===============
+                for counters, var in enumerate(['flowrate', 'velocity', 'depth', 'nudge']):
+                    QVD = ncfile.createVariable(
+                        varname=var,
+                        datatype=np.float32,
+                        dimensions=('feature_id', 'time_step (sec)',),
+                    )
+
+                    QVD.units = 'm3/s m/s m m3/s'
+                    QVD.description = f'Data for {var}'
+                    
+                    # Prepare data for writing
+                    data_array = subset_df.iloc[:, counters::4].to_numpy(dtype=np.float32)
+                    
+                    # Set data for each feature_id and time_step
+                    ncfile.variables[var][:] = data_array
+                feature_id = ncfile.createVariable(
+                    varname='feature_id',
+                    datatype=np.int32,
+                    dimensions=('feature_id',),
+                )
+                feature_id[:] = flowveldepth.index.to_numpy(dtype=np.int32)
+                feature_id.units = 'None'
+                feature_id.description = 'Feature IDs'
+                ###
+                time_step = ncfile.createVariable(
+                    varname='time_step (sec)',
+                    datatype=np.int32,
+                    dimensions=('time_step (sec)',),
+                )
+                time_step[:] = np.array(time_dim[:subset_df.iloc[:, 0::4].shape[1]], dtype=np.int32)
+                time_step.units = 'sec'
+                time_step.description = 'time stamp'
+                # =========== GLOBAL ATTRIBUTES ===============
+                ncfile.setncatts(
+                    {
+                        'TITLE': 'OUTPUT FROM T-ROUTE',
+                        'Time step (sec)': f'{stream_output_internal_frequency}',
+                        'model_initialization_time': t0.strftime('%Y-%m-%d_%H:%M:%S'),
+                        'model_reference_time': time_steps[counter].strftime('%Y-%m-%d_%H:%M:%S'),
+                        'comment': f'The file includes {stream_output_timediff} hour data which includes {len(time_dim)} timesteps',
+                        'code_version': '',
+                    }
+                )
+                LOG.debug(f"Flowveldepth data saved as NetCDF files in {stream_output_directory}")
+
 def write_flowveldepth_netcdf(stream_output_directory, 
                               flowveldepth, 
                               nudge, 
                               usgs_positions_id, 
                               t0, 
                               stream_output_timediff, 
                               stream_output_type,
-                              stream_output_internal_frequency = 5):
+                              stream_output_internal_frequency = 5,
+                              cpu_pool = 6, 
+                              use_parallel = True):
```

Comment:
It looks like this `use_parallel` has not been added to output.py when the function gets called? If you want to do it this way, you also need to determine when/how `use_parallel` is determined based on user input. I think you might be able to just drop this though and use only `cpu_pool` in the `if/else` statements. What do you think?

Reply:
Yes sure. So I can just say if cpu_pul > 1 do parallel otherwise don't. The only caveat is user have to set cpu_pool = 1 in output.py not here (the value 6 here doesn't matter)

Reply:
cpu_pool is already an input into `nwm_output_generator` in [output.py](https://github.com/shorvath-noaa/t-route/blob/4cd3e498ca3e44941ca93401209ffc791f548b6a/src/troute-nwm/src/nwm_routing/output.py#L49-L65), so it just needs to be added as in input into [nhd_io.write_flowveldepth_netcdf()](https://github.com/shorvath-noaa/t-route/blob/4cd3e498ca3e44941ca93401209ffc791f548b6a/src/troute-nwm/src/nwm_routing/output.py#L205-L212)